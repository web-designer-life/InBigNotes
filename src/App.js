import React from 'react';
import './App.css';
import Container from './Container';

class App extends React.Component {
  render() {
      return (
          <div>
              <Container />
          </div>
      );
  }
}

export default App;
