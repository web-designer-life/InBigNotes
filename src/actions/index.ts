const FETCH_NOTES_PENDING = 'FETCH_NOTES_PENDING';
const FETCH_NOTES_SUCCESS = 'FETCH_NOTES_SUCCESS';
const FETCH_NOTES_FAIL = 'FETCH_NOTES_FAIL';

const FETCH_NOTE_PENDING = 'FETCH_NOTE_PENDING';
const FETCH_NOTE_SUCCESS = 'FETCH_NOTE_SUCCESS';
const FETCH_NOTE_FAIL = 'FETCH_NOTE_FAIL';

const ADD_NOTE_PENDING = 'ADD_NOTE_PENDING';
const ADD_NOTE_SUCCESS = 'ADD_NOTE_SUCCESS';
const ADD_NOTE_FAIL = 'ADD_NOTE_FAIL';

const UPDATE_NOTE_PENDING = 'UPDATE_NOTE_PENDING';
const UPDATE_NOTE_SUCCESS = 'UPDATE_NOTE_SUCCESS';
const UPDATE_NOTE_FAIL = 'UPDATE_NOTE_FAIL';

const DELETE_NOTES_PENDING = 'DELETE_NOTES_PENDING';
const DELETE_NOTES_SUCCESS = 'DELETE_NOTES_SUCCESS';
const DELETE_NOTES_FAIL = 'DELETE_NOTES_FAIL';

const ADD_SELECTED_NOTE = 'ADD_SELECTED_NOTE';
const REMOVE_UNSELECTED_NOTE = 'REMOVE_UNSELECTED_NOTE';

const SELECT_ALL_NOTES = 'SELECT_ALL_NOTES';
const UNSELECT_ALL_NOTES = 'UNSELECT_ALL_NOTES';

const FILTER = 'FILTER';

const NAVIGATE_TO_PAGE = 'NAVIGATE_TO_PAGE';

const RESET_STORE = 'RESET_STORE';

export default {
    FETCH_NOTES_PENDING,
    FETCH_NOTES_SUCCESS,
    FETCH_NOTES_FAIL,
    FETCH_NOTE_PENDING,
    FETCH_NOTE_SUCCESS,
    FETCH_NOTE_FAIL,
    ADD_NOTE_PENDING,
    ADD_NOTE_SUCCESS,
    ADD_NOTE_FAIL,
    UPDATE_NOTE_PENDING,
    UPDATE_NOTE_SUCCESS,
    UPDATE_NOTE_FAIL,
    DELETE_NOTES_PENDING,
    DELETE_NOTES_SUCCESS,
    DELETE_NOTES_FAIL,
    ADD_SELECTED_NOTE,
    REMOVE_UNSELECTED_NOTE,
    SELECT_ALL_NOTES,
    UNSELECT_ALL_NOTES,
    FILTER,
    NAVIGATE_TO_PAGE,
    RESET_STORE,
};
