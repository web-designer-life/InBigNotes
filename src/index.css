body {
    font-family: 'Lora', 'Arial', sans-serif;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    height: 100%;
    margin: 0;
    padding: 0;
}

code {
    font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New', monospace;
}

#root {
    display: flex;
    flex-direction: column;
    max-width: 1280px;
    height: 100%;
    margin: auto;
}
