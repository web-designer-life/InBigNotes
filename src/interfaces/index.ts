export interface INote {
    id: string;
    title: string;
    text: string;
    created_at: number;
    updated_at: number;
};
