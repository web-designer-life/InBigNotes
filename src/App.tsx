import * as React from 'react';
import { GlobalStyle } from './Components/Style/GlobalStyle';
import Container from './Container';
import './App.css';

class App extends React.Component {
    render() {
        return (
            <>
                <GlobalStyle />
                <Container />
            </>
        );
    }
};
  
export default App;
