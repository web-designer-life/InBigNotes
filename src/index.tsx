import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import history, { store } from './store/store';
import App from './containers/App';

ReactDOM.render(
    <React.StrictMode>
        <Provider store={store}>
            <App history={history} />
        </Provider>
    </React.StrictMode>,
    document.getElementById('root')
);
