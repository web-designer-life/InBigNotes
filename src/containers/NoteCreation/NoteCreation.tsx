import * as React from 'react';
import { Link } from 'react-router-dom';
import Note from '../../components/Note/Note';
import { Loader } from '../../components/Loader/Loader';
import { fetchNotes, addNote, deleteNotes, addSelectedNote, removeUnselectedNote, selectAllNotes, unselectAllNotes } from '../../store/actions/notes';
import { Note as INote } from '../../interfaces';

interface Props {
    isLoading: Boolean,
    addNoteAction(notes: INote[]): Function,
};

export default class NoteCreation extends React.Component<Props> {
    componentDidMount() {
        this.props.fetchNotesAction();
    }

    render() {
        const { 
            isLoading, 
        } = this.props;

        return (
            <>
                <Loader isLoading={isLoading}/>
                <Note />
            </>
        );
    }
};
