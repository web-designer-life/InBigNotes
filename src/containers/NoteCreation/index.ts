import { connect } from 'react-redux';
import NoteCreation from './NoteCreation';
import { Note } from '../../interfaces';
import { addNoteData } from '../../actionCreators/note';
import { navigateToPage, resetStore } from '../../actionCreators/common';

const mapStateToProps = (store: any) => {
    return {
        isLoading: store.note.isLoading,
        error: store.note.error,
    }
};

const mapDispatchToProps = (dispatch: any) => {
    return {
        addNoteAction: (note: Note) => dispatch(addNoteData(note)),
        navigateToPageAction: (path: string) => dispatch(navigateToPage(path)),
        resetStoreAction: () => dispatch(resetStore()),
    }
};
 
export default connect(
    mapStateToProps,
    mapDispatchToProps,
)(NoteCreation);
