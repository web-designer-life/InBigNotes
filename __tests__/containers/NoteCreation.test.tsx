import React from 'react';
import { mount, shallow } from 'enzyme';
import NoteCreation from '../../src/containers/NoteCreation/NoteCreation';
import { INote } from '../../src/interfaces';

const noteProps = {
    isLoading: false,
    error: false,
    addNoteAction: (note: INote) => Function,
    navigateToPageAction: (path: string) => Function,
    resetStoreAction: () => Function,
};

const notePropsWithLoading = {
    isLoading: true,
    error: false,
    addNoteAction: (note: INote) => Function,
    navigateToPageAction: (path: string) => Function,
    resetStoreAction: () => Function,
};

const notePropsWithError = {
    isLoading: false,
    error: true,
    addNoteAction: (note: INote) => Function,
    navigateToPageAction: (path: string) => Function,
    resetStoreAction: () => Function,
};

describe('NoteView', () => {
    it('should render NoteView', () => {
        const component = shallow(<NoteCreation {...noteProps} />);

        expect(component).toMatchSnapshot();
    });

    it('should render Loader', () => {
        const component = shallow(<NoteCreation {...notePropsWithLoading} />);

        expect(component).toMatchSnapshot();

        component.unmount();
    });

    describe('handlers', () => {
        describe('handleNavigateToHome', () => {
            it('should call method', () => {
                const component = mount(<NoteCreation {...notePropsWithError} />);

                const createNoteButton = component.find('button[color="green"]').first();
                createNoteButton.simulate('click');

                expect(component).toMatchSnapshot();
            });
        });
    });
});
