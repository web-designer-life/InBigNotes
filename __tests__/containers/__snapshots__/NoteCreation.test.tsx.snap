// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteView handlers handleNavigateToHome should call method 1`] = `
<NoteCreation
  addNoteAction={[Function]}
  error={true}
  isLoading={false}
  navigateToPageAction={[Function]}
  resetStoreAction={[Function]}
>
  <Error
    navigateToPage={[Function]}
  >
    <styled.div>
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-VigVT",
              "isStatic": false,
              "lastClassName": "heZnKW",
              "rules": Array [
                "
	display: flex;
	justify-content: center;
	align-items: center;
	height: 100vh;
	flex-direction: column;
",
              ],
            },
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-VigVT",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
      >
        <div
          className="sc-VigVT heZnKW"
        >
          <styled.h1>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-jTzLTM",
                    "isStatic": false,
                    "lastClassName": "fgxopO",
                    "rules": Array [
                      "
    margin: 10px;
",
                    ],
                  },
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-jTzLTM",
                  "target": "h1",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <h1
                className="sc-jTzLTM fgxopO"
              >
                404 - Not Found!
              </h1>
            </StyledComponent>
          </styled.h1>
          <Button
            color="green"
            disabled={false}
            onClick={[Function]}
            text="Go home"
            type="button"
          >
            <styled.button
              color="green"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <StyledComponent
                color="green"
                disabled={false}
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bdVaJa",
                      "isStatic": false,
                      "lastClassName": "mScxf",
                      "rules": Array [
                        "
    min-width: 170px;
    border-radius: 15px;
    color: ",
                        "#FFFFFF",
                        ";
    padding: 10px 25px;
    transition-duration: .1s;
    font-style: normal;
    font-weight: bold;
    font-size: 24px;
    &:hover {
        background-color: ",
                        "#FFFFFF",
                        ";
    }
    &:disabled {
        opacity: .5;
        border: 4px solid ",
                        "#808080",
                        ";
        background-color: ",
                        "#808080",
                        ";
        &:hover { 
            background-color: ",
                        "#808080",
                        ";
            color: ",
                        "#FFFFFF",
                        ";
            cursor: default;
        }
    }
    ",
                        [Function],
                        "
    ",
                        [Function],
                        "
    @media (max-width: 450px) {
        min-width: auto;
        width: 100%;
    }
",
                      ],
                    },
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bdVaJa",
                    "target": "button",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                onClick={[Function]}
                type="button"
              >
                <button
                  className="sc-bdVaJa mScxf"
                  color="green"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  Go home
                </button>
              </StyledComponent>
            </styled.button>
          </Button>
        </div>
      </StyledComponent>
    </styled.div>
  </Error>
</NoteCreation>
`;

exports[`NoteView should render Loader 1`] = `<Loader />`;

exports[`NoteView should render NoteView 1`] = `
<Note
  addOrUpdateNote={[Function]}
  navigateToPage={[Function]}
  typeName="Save"
/>
`;
